### PR TITLE
Adding function: Encrypt, Decrypt

### PR DESCRIPTION
**Pull request by librone4u**

### CODE REVIEW

File: src/Service/DecryptService.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        try(BufferedReader reader = new BufferedReader(new FileReader(inputStream));
+            BufferedWriter writer = new BufferedWriter(new FileWriter(outputStream))) {
+
+            String line;
+            while((line = reader.readLine()) != null){
+                String text = textTransformer.moveLetterOnLeftPosition(key, line);
+                writer.write(text);
+                writer.newLine();
+            }
+
```
Comment (Volodymyr95) [inline src/Service/DecryptService.java]: delete empty lines

File: src/Service/DecryptService.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```
Comment (Volodymyr95) [inline src/Service/DecryptService.java]: do u have in code some exception handler for throwing it?

File: src/Service/EncryptService.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        try(BufferedReader reader = new BufferedReader(new FileReader(inputStream));
+        BufferedWriter writer = new BufferedWriter(new FileWriter(outputStream))){
+
+        String line;
+        while((line = reader.readLine()) != null){
+            String text = textTransformer.moveLetterOnRightPosition(key, line);
+            writer.write(text);
+            writer.newLine();
+        }
+
```
Comment (Volodymyr95) [inline src/Service/EncryptService.java]: delete empty lines

File: src/Service/EncryptService.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
```
Comment (Volodymyr95) [inline src/Service/EncryptService.java]: delete empty lines

File: src/Service/EncryptService.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        String line;
+        while((line = reader.readLine()) != null){
+            String text = textTransformer.moveLetterOnRightPosition(key, line);
+            writer.write(text);
+            writer.newLine();
+        }
+
+
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
```
Comment (Volodymyr95) [inline src/Service/EncryptService.java]: don't throw exception inside catch block, u don't have handler for it. Add informative message here

File: src/SupportClasses/CreateNewFile.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package SupportClasses;
+
+import java.io.File;
+import java.io.IOException;
+
+public class CreateNewFile {
```
Comment (Volodymyr95) [inline src/SupportClasses/CreateNewFile.java]: don't use verbs inside class name

File: src/Transformer/MapAlphabet.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.Map;
+
+public class MapAlphabet {
+    final private String lowerCaseAlphabet = "abcdefghijklmnopqrstuvwxyz";
+    final private String upperCaseAlphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    //modes: right|left
+    public Map<Character, Character> getMapAlphabet(int key, String mode){
+        Map<Character, Character> transformedAlphabet = new LinkedHashMap<>();
+
+        if(mode.equalsIgnoreCase("right")) {
```
Comment (Volodymyr95) [inline src/Transformer/MapAlphabet.java]: make "right" constant

File: src/Transformer/MapAlphabet.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            }
+            //adding to map upper case letter
+            for (int i = 0; i < upperCaseAlphabet.length(); i++) {
+                int shiftedIndex = (i + key) % upperCaseAlphabet.length();
+                char originalChar = upperCaseAlphabet.charAt(i);
+                char shiftedChar = upperCaseAlphabet.charAt(shiftedIndex);
+                transformedAlphabet.put(originalChar, shiftedChar);
+            }
+
+        } else if (mode.equalsIgnoreCase("left")) {
```
Comment (Volodymyr95) [inline src/Transformer/MapAlphabet.java]: make left constan

File: src/Service/DecryptService.java (hunk 2 above)
Comment (librone4u) [reply src/Service/DecryptService.java]: nope, im goint to do this later

File: src/Service/DecryptService.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+import SupportClasses.FileCreator;
+import SupportClasses.Mode;
+import Transformer.TextTransformer;
+
+import java.io.*;
+
+public class DecryptService {
+    TextTransformer textTransformer = new TextTransformer();
+    FileCreator createNewFile = new FileCreator();
```
Comment (Volodymyr95) [inline src/Service/DecryptService.java]: createNewFile to fileCreator

File: src/Service/EncryptService.java [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+import SupportClasses.FileCreator;
+import SupportClasses.Mode;
+import Transformer.TextTransformer;
+
+import java.io.*;
+
+public class EncryptService {
+    private TextTransformer textTransformer = new TextTransformer();
+    private FileCreator createNewFile = new FileCreator();
+    public void encrypt(String filePath, String key){
```
Comment (Volodymyr95) [inline src/Service/EncryptService.java:13]: add empty line between method and variables

File: src/SupportClasses/FileCreator.java [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    public File createNewFile(File file, Enum mode){
+        String currentFileName = file.getName();
+        int dotIndex = currentFileName.lastIndexOf(".");
+        int openParenIndex = currentFileName.lastIndexOf("(");
+
+        if(openParenIndex != -1){
+            currentFileName = currentFileName.substring(0 , openParenIndex) + currentFileName.substring(dotIndex);
+            dotIndex = currentFileName.lastIndexOf(".");
+        }
+
```
Comment (Volodymyr95) [inline src/SupportClasses/FileCreator.java:17]: refactoring, delete empty line here

File: src/SupportClasses/Runner.java [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+import Service.EncryptService;
+
+public class Runner {
+    public void run(String[] arguments){
+        EncryptService encryptService = new EncryptService();
+        DecryptService decryptService = new DecryptService();
+        if(arguments[0].equals(Constants.ENCRYPT_MODE)){
+            encryptService.encrypt(arguments[1], arguments[2]);
+        }
+        else if(arguments[0].equals(Constants.DECRYPT_MODE)){
```
Comment (Volodymyr95) [inline src/SupportClasses/Runner.java]: 0,1,2 - make it variables

File: src/Service/DecryptService.java [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+        try(BufferedReader reader = new BufferedReader(new FileReader(inputStream));
+            BufferedWriter writer = new BufferedWriter(new FileWriter(outputStream))) {
+
+            String line;
+            while((line = reader.readLine()) != null){
+                String text = textTransformer.moveLetterOnLeftPosition(Integer.parseInt(key), line);
+                writer.write(text);
+                writer.newLine();
+            }
+        } catch (Exception e) {
```
Comment (Volodymyr95) [inline src/Service/DecryptService.java]: why exception? catch checked exceptions from this block

File: src/Service/EncryptService.java [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+        try(BufferedReader reader = new BufferedReader(new FileReader(inputStream));
+        BufferedWriter writer = new BufferedWriter(new FileWriter(outputStream))){
+
+        String line;
+        while((line = reader.readLine()) != null){
+            String text = textTransformer.moveLetterOnRightPosition(Integer.parseInt(key), line);
+            writer.write(text);
+            writer.newLine();
+        }
+        } catch (Exception e) {
```
Comment (Volodymyr95) [inline src/Service/EncryptService.java]: why exception? catch checked exceptions from this block

File: src/SupportClasses/FileCreator.java [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+        final String absolutePath = file.getAbsolutePath();
+        String nameForNewFiles = null;
+
+        if(dotIndex != -1){
+            nameForNewFiles = currentFileName.substring(0, dotIndex) + "(" + mode + ")" + currentFileName.substring(dotIndex);
+        }
+        File newFile = new File(absolutePath.substring(0, absolutePath.lastIndexOf(File.separator) + 1) + nameForNewFiles);
+        try {
+            newFile.createNewFile();
+        } catch (Exception e) {
```
Comment (Volodymyr95) [inline src/SupportClasses/FileCreator.java]: why exception? catch checked exceptions from this block

File: src/SupportClasses/Runner.java [hunk 16]
```diff
@@ -0,0 +1,24 @@
+package SupportClasses;
+
+import Service.DecryptService;
+import Service.EncryptService;
+
+import java.io.File;
+
+public class Runner {
+    public void run(String[] arguments){
+        final String MODE = arguments[0];
```
Comment (Volodymyr95) [inline src/SupportClasses/Runner.java]: 0,1,2 are magic numbers

File: src/Service/DecryptService.java [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+            BufferedWriter writer = new BufferedWriter(new FileWriter(outputStream))) {
+
+            String line;
+            while((line = reader.readLine()) != null){
+                String text = textTransformer.moveLetterOnLeftPosition(Integer.parseInt(key), line);
+                writer.write(text);
+                writer.newLine();
+            }
+        } catch (IOException e) {
+            System.out.println("File not found");
```
Comment (Volodymyr95) [inline src/Service/DecryptService.java]: System.err.println

File: src/Service/EncryptService.java [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+        BufferedWriter writer = new BufferedWriter(new FileWriter(outputStream))){
+
+        String line;
+        while((line = reader.readLine()) != null){
+            String text = textTransformer.moveLetterOnRightPosition(Integer.parseInt(key), line);
+            writer.write(text);
+            writer.newLine();
+            }
+        } catch (IOException e) {
+            System.out.println("File not found");;
```
Comment (Volodymyr95) [inline src/Service/EncryptService.java]: System.err.println

File: src/SupportClasses/FileCreator.java [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+        String nameForNewFiles = null;
+
+        if(dotIndex != -1){
+            nameForNewFiles = currentFileName.substring(0, dotIndex) + "(" + mode + ")" + currentFileName.substring(dotIndex);
+        }
+        File newFile = new File(absolutePath.substring(0, absolutePath.lastIndexOf(File.separator) + 1) + nameForNewFiles);
+        try {
+            newFile.createNewFile();
+        } catch (IOException e) {
+            System.out.println("Path is wrong");
```
Comment (Volodymyr95) [inline src/SupportClasses/FileCreator.java]: System.err.println

File: src/SupportClasses/Validator.java [hunk 20]
```diff
@@ -0,0 +1,35 @@
+package SupportClasses;
+import Exception.InvalidArgumentException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Calendar;
+
+public class Validator {
+    public static void ValidateInputData(String[] args){
+        ValidateLengthOfArguments(args);
```
Comment (Volodymyr95) [inline src/SupportClasses/Validator.java]: method from small letter!

File: src/SupportClasses/Validator.java [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+        ValidateCommand(args[Constants.MODE_INDEX]);
+        ValidateFilePath(args[Constants.FILE_PATH_INDEX]);
+    }
+    private static void ValidateLengthOfArguments(String[] args){
+        if(args.length > 2){
```
Comment (Volodymyr95) [inline src/SupportClasses/Validator.java]: what 2 means?

Comment (librone4u) [reply src/SupportClasses/Validator.java]: its means that argument cannot be more than 2, cause we have only 3 arguments(command filepath and key), or im wrong?

**Pull request merged**